force-app/main/default/classes/CON_DeleteContactOverrideSelector.cls

```diff
@@ -0,0 +1,133 @@
+/*
+    Copyright (c) 2021, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @group Contacts
+* @group-content ../../ApexDocContent/Contacts.htm
+* @description Selects records for the Contact Delete button override.
+*/
+public with sharing class CON_DeleteContactOverrideSelector {
+
+    public Contact queryContact(Id contactId) {
+        Contact contactDetails = [
+            SELECT Id, Name, FirstName, LastName,
+                AccountId, Account.Name, Account.npe01__SYSTEMIsIndividual__c
+            FROM Contact
+            WHERE Id = :contactId
+        ];
+        return contactDetails;
+    }
+
+    public Boolean isAccountWithSingleContact(Id accountId) {
+        List<AggregateResult> contactsInAccount = [
+            SELECT COUNT(id) ct
+            FROM Contact
+            WHERE AccountId = :accountId
+            GROUP BY AccountId
+            HAVING COUNT(id) = 1
+        ];
+
+        if (contactsInAccount.size() != 1 || !(contactsInAccount[0].get('ct') == 1)) {
+            return false;
+        }
+        return true;
+    }
+
+    /*********************************************************************************************************
+    * @description Check if the running user has minimum access to delete the record
+    * @return Boolean
+    */
+    public Boolean canDeleteRecord(Id recordId) {
```

Comment:
The permissions checks should not be in the selector class.  In order to maintain separation of concerns they should be split out into their own class.